src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1306,6 +1310,22 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def listPublishedWorkspaces() = {
+    asFCAdmin {
+      dataSource.inTransaction { dataAccess =>
+        dataAccess.workspaceQuery.listPublished.map(RequestComplete(StatusCodes.OK, _))
+      }
+    }
+  }
+
+  def listWorkspacesWithAttribute(attributeMap: AttributeMap): Future[PerRequestMessage] = {
+    asFCAdmin {
+      dataSource.inTransaction { dataAccess =>
```

Comment:
if we are only supporting size=1, it should not be a map at all, just pass in the key and the value.